### PR TITLE
Bao/add locker tests

### PR DESCRIPTION
**Pull request by StarryFolf**
test: add tests for add locker

### CODE REVIEW

File: src/Api/Controllers/LockersController.cs [hunk 1]
```diff
@@ -1,14 +1,22 @@
...
+using Application.Lockers.Commands.RemoveLocker;
 using Microsoft.AspNetCore.Mvc;
 
 namespace Api.Controllers;
 
 public class LockersController : ApiControllerBase
 {
     [HttpPost]
-    public async Task<ActionResult<Result<LockerDto>>> AddLocker([FromBody] AddLockerCommand command)
+    public async Task<ActionResult<Result<LockerDto>>> AddLocker([FromBody] CreateLockerCommand command)
```
Comment (ChienNQuang) [inline src/Api/Controllers/LockersController.cs:12]: Why you leave 2 changes in 2 different PR?

Comment (ChienNQuang) [reply src/Api/Controllers/LockersController.cs:12]: and this change is completely out of no where in context of branch

File: tests/Application.Tests.Integration/Lockers/Commands/CreateLockerTests.cs [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+public class CreateLockerTests : BaseClassFixture
+{
+    private static IServiceScopeFactory _scopeFactory = null!;
+    public CreateLockerTests(CustomApiFactory apiFactory) : base(apiFactory)
+    {
+        _scopeFactory = apiFactory.Services.GetRequiredService<IServiceScopeFactory>();
+    }
+
+    [Fact]
+    public async Task ShouldReturnLocker_WhenLockerDetailsAreValid()
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Lockers/Commands/CreateLockerTests.cs:23]: WhenCreateDetails to be consistent with other tests

File: tests/Application.Tests.Integration/Lockers/Commands/CreateLockerTests.cs [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+
+        var createLockerCommand = new CreateLockerCommand()
+        {
+            Name = new Faker().Name.JobTitle(),
+            Description = new Faker().Lorem.Sentence(),
+            Capacity = 1,
+            RoomId = room.Id,
+        };
+
+        room.NumberOfLockers += 1;
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Lockers/Commands/CreateLockerTests.cs:46]: this is unnecessary

File: tests/Application.Tests.Integration/Lockers/Commands/CreateLockerTests.cs [hunk 4]
```diff
@@ -0,0 +1,235 @@
...
+        // Act
+        
+        var locker = await SendAsync(createLockerCommand);
+        
+        // Assert
+        locker.Name.Should().Be(createLockerCommand.Name);
+        locker.Description.Should().Be(createLockerCommand.Description);
+        locker.Capacity.Should().Be(createLockerCommand.Capacity);
+        locker.Room.Id.Should().Be(room.Id);
+
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Lockers/Commands/CreateLockerTests.cs:57]: Lack one assert of locker's number of lockers in room must increase by 1

File: tests/Application.Tests.Integration/Lockers/Commands/CreateLockerTests.cs [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+        
+        var createLockerCommand2 = new CreateLockerCommand()
+        {
+            Name = createLockerCommand.Name,
+            Description = new Faker().Lorem.Sentence(),
+            Capacity = 1,
+            RoomId = room.Id,
+        };
+
+        room.NumberOfLockers += 1;
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Lockers/Commands/CreateLockerTests.cs:99]: this is not needed

File: tests/Application.Tests.Integration/Lockers/Commands/CreateLockerTests.cs [hunk 6]
```diff
@@ -0,0 +1,235 @@
...
+
+        var createLockerCommand = new CreateLockerCommand()
+        {
+            Name = new Faker().Name.JobTitle(),
+            Description = new Faker().Lorem.Sentence(),
+            Capacity = 1,
+            RoomId = room.Id,
+        };
+        
+        room.NumberOfLockers += 1;
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Lockers/Commands/CreateLockerTests.cs:89]: this is not needed

File: tests/Application.Tests.Integration/Lockers/Commands/CreateLockerTests.cs [hunk 7]
```diff
@@ -0,0 +1,235 @@
...
+        
+        //Cleanup
+        var lockerEntity = await FindAsync<Locker>(locker.Id);
+        var roomEntity = await FindAsync<Room>(room.Id);
+        Remove(lockerEntity);
+        Remove(roomEntity);
+    }
+
+    [Fact]
+    public async Task ShouldReturnLocker_WhenLockersHasSameNameButInDifferentRooms()
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Lockers/Commands/CreateLockerTests.cs:117]: This test is unnecessary, the above one is sufficient already

File: tests/Application.Tests.Integration/Lockers/Commands/CreateLockerTests.cs [hunk 8]
```diff
@@ -0,0 +1,235 @@
...
+
+        var createLockerCommand = new CreateLockerCommand()
+        {
+            Name = new Faker().Name.JobTitle(),
+            Description = new Faker().Lorem.Sentence(),
+            Capacity = 1,
+            RoomId = room.Id,
+        };
+        
+        room.NumberOfLockers += 1;
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Lockers/Commands/CreateLockerTests.cs:210]: this too is not needed

File: tests/Application.Tests.Integration/Lockers/Commands/CreateLockerTests.cs [hunk 9]
```diff
@@ -0,0 +1,235 @@
...
+        
+        var createLockerCommand2 = new CreateLockerCommand()
+        {
+            Name = createLockerCommand.Name,
+            Description = new Faker().Lorem.Sentence(),
+            Capacity = 1,
+            RoomId = room.Id,
+        };
+
+        room.NumberOfLockers += 1;
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Lockers/Commands/CreateLockerTests.cs:220]: this too is not needed

**Review by ChienNQuang — CHANGES_REQUESTED** [review]
Very good but need some adjustment

**Review by ChienNQuang — CHANGES_REQUESTED** [review]
Very good but need some adjustment

File: tests/Application.Tests.Integration/Lockers/Commands/AddLockerTests.cs [hunk 10]
```diff
@@ -0,0 +1,230 @@
...
+
+        var createLockerCommand = new AddLockerCommand()
+        {
+            Name = new Faker().Name.JobTitle(),
+            Description = new Faker().Lorem.Sentence(),
+            Capacity = 1,
+            RoomId = room.Id,
+        };
+        
+        var createLockerCommand2 = new AddLockerCommand()
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Lockers/Commands/AddLockerTests.cs:91]: This is unnecessary, using the same command

File: tests/Application.Tests.Integration/Lockers/Commands/AddLockerTests.cs [hunk 11]
```diff
@@ -0,0 +1,230 @@
...
+        {
+            Name = createLockerCommand.Name,
+            Description = new Faker().Lorem.Sentence(),
+            Capacity = 1,
+            RoomId = room.Id,
+        };
+        
+        // Act
+        
+        var locker = await SendAsync(createLockerCommand);
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Lockers/Commands/AddLockerTests.cs:101]: This should not be in Act

File: tests/Application.Tests.Integration/Lockers/Commands/AddLockerTests.cs [hunk 12]
```diff
@@ -0,0 +1,230 @@
...
+        
+        //Cleanup
+        var lockerEntity = await FindAsync<Locker>(locker.Id);
+        var roomEntity = await FindAsync<Room>(room.Id);
+        Remove(lockerEntity);
+        Remove(roomEntity);
+    }
+
+    [Fact]
+    public async Task ShouldReturnLocker_WhenLockersHasSameNameButInDifferentRooms()
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Lockers/Commands/AddLockerTests.cs:115]: This test still is not necessary

File: tests/Application.Tests.Integration/Lockers/Commands/AddLockerTests.cs [hunk 13]
```diff
@@ -0,0 +1,230 @@
...
+        var createLockerCommand2 = new AddLockerCommand()
+        {
+            Name = createLockerCommand.Name,
+            Description = new Faker().Lorem.Sentence(),
+            Capacity = 1,
+            RoomId = room.Id,
+        };
+        
+        // Act
+        var locker = await SendAsync(createLockerCommand);
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Lockers/Commands/AddLockerTests.cs:218]: This too should not be in Act

File: tests/Application.Tests.Integration/Lockers/Commands/DisableLockerTests.cs [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+            Name = new Faker().Commerce.ProductName(),
+            Description = new Faker().Lorem.Sentence(),
+            Capacity = 1,
+            IsAvailable = true,
+            NumberOfLockers = 0,
+        };
+
+        await AddAsync(room);
+
+        var createLockerCommand = new AddLockerCommand()
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Lockers/Commands/DisableLockerTests.cs:33]: createCommand = addCommand?

File: tests/Application.Tests.Integration/Lockers/Commands/DisableLockerTests.cs [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+
+public class DisableLockerTests : BaseClassFixture
+{
+    public DisableLockerTests(CustomApiFactory apiFactory) : base(apiFactory)
+    {
+        
+    }
+
+    [Fact]
+    public async Task ShouldRemoveLocker_WhenLockerExists()
```
Comment (ChienNQuang) [inline tests/Application.Tests.Integration/Lockers/Commands/DisableLockerTests.cs:18]: This should be named ShouldDisableLocker_WhenLockerExistsAndIsAvailable

**Review by ChienNQuang — CHANGES_REQUESTED** [review]
Should add one more test in DisableLockerTests: Should do what when the locker is already disabled?

File: src/Application/Lockers/Commands/DisableLocker/DisableLockerCommand.cs [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+        var folders = _context.Folders.Where(x => x.Locker.Room.Id.Equals(locker.Id));
+        
+        foreach (var folder in folders)
+        {
+            folder.IsAvailable = false;
+        }
+        _context.Folders.UpdateRange(folders);
+        
+        locker.IsAvailable = false;
+        locker.Room.NumberOfLockers -= 1;
```
Comment (kaitoz11) [inline src/Application/Lockers/Commands/DisableLocker/DisableLockerCommand.cs:59]: What should we do? @ChienNQuang @Vzart 
Keep or not?

**Review by Vzart — APPROVED** [review]
LGTM!

**Review by ChienNQuang — APPROVED** [review]

**Pull request merged**